src/components/organisms/notifier/NotificationChannelsList.tsx

```diff
@@ -46,12 +43,14 @@ const NotificationChannelsList: FC<Props> = ({ channels }) => {
     if (index === -1) addedChannels.push(notifierChannel)
     else addedChannels[index].destination = notifierChannel.destination
     const newChannels = [...addedChannels]
```

Comment:
Could we change this this time around please?
```suggestion
    const existingChannel = addedChannels.find(({ type }) => type === notifierChannel.type)
    const newChannels = [...addedChannels]

    if (existingChannel) {
       newChannels.push(notifierChannel)
    } else {
       newChannels.destination = notifierChannel.destination
```
Especially the mutating state bit. Perhaps that is why you needed two states to hold your channels? now you can remove one as they are set together to the same values there is no need to have them both. Or am I missing something?

Reply:
This was the main reason I requested changes. Please do not mutate state.
Work off the cloned array from the start of the function. Also once again don't use index === -1, please. First, you don't need to use the index here, just get the object instead. then mutate the object. const existingChannel = channelsClone.find(({type}) => type === notifierChannel.type) will give you the object within the array. you don't need to access it by index.